ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/nulls/NullIfTest.java

```diff
@@ -0,0 +1,38 @@
+package io.confluent.ksql.function.udf.nulls;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+public class NullIfTest {
+
+  private NullIf udf;
+
+  @Before
+  public void setUp() {
+    udf = new NullIf();
+  }
+
+  @Test
+  public void shouldReturnNullIfBothValuesAreNulls() {
+    assertThat(udf.nullIf(null, null), is(nullValue()));
+  }
+
+  @Test
+  public void shouldReturnNullIfValue1IsNull() {
+    assertThat(udf.nullIf(null, "a"), is(nullValue()));
+  }
+
+  @Test
+  public void shouldReturnNullIfBothValuesAreEqual() {
+    assertThat(udf.nullIf("a", "a"), is(nullValue()));
+  }
+
+  @Test
+  public void shouldReturnValue1IfBothValuesAreNonEqual() {
+    assertThat(udf.nullIf("a", "b"), is("a"));
+  }
```

Comment:
Could you add a test when value2 is null and value1 is not null?
Could you add a test to verify case-sensitive comparison? `nullif("a", "A")`

Reply:
Added more tests